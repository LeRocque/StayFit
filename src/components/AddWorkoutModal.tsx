import {
  ChangeEvent,
  FormEvent,
  useState,
  useTransition,
} from "react";
import { WorkoutModalProps } from "../frontendTypes";

export const AddWorkoutModal = ({
  userId,
  handleWorkoutModal,
}: WorkoutModalProps) => {
  const [workoutName, setWorkoutName] = useState("");
  const [muscleTarget, setMuscleTarget] = useState("Back");
  const [weight, setWeight] = useState("");
  const [reps, setReps] = useState("");
  const [isPending, startTransition] = useTransition();
  const [errorMessage, setErrorMessage] = useState<string | null>(null);

  const handleSubmit = async (e: FormEvent): Promise<void> => {
    e.preventDefault();
    try {
      const response = await fetch("/workout/add", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          user_id: userId,
          workoutName: workoutName,
          muscleTarget: muscleTarget,
          weight: weight,
          reps: reps,
        }),
      });
      if (response.ok) {
        setWorkoutName("");
        setMuscleTarget("");
        setWeight("");
        setReps("");
        startTransition(() => {
          handleWorkoutModal();
        });
      } else {
        alert("invalid input");
      }
    } catch (err) {
      console.error(err);
      setErrorMessage("An error occuring while attempting to add new workout");
    }
  };

  const handleModalKeyPress = (
    e: React.KeyboardEvent<HTMLDivElement>,
  ): void => {
    if (e.key === "Enter") {
      handleWorkoutModal();
    }
  };

  const handleWorkoutName = (e: ChangeEvent<HTMLInputElement>): void =>
    setWorkoutName(e.target.value);
  const handleMuscleTarget = (e: ChangeEvent<HTMLSelectElement>): void =>
    setMuscleTarget(e.target.value);
  const handleWeight = (e: ChangeEvent<HTMLInputElement>): void =>
    setWeight(e.target.value);
  const handleReps = (e: ChangeEvent<HTMLInputElement>): void =>
    setReps(e.target.value);

  return (
    <div
      className="position-fixed fixed top-0 left-0 z-50 flex h-full w-full items-center justify-center bg-opacity-90 bg-gradient-to-br from-slate-200 via-slate-500 to-slate-700 opacity-90"
      id="modal-container"
      onKeyDown={handleModalKeyPress}
      role="button"
      tabIndex={0}
    >
      {errorMessage && (
        <div className="error-message">
          {errorMessage}
          <button
            className="close-button"
            onClick={() => setErrorMessage(null)}
          >
            Dismiss Error
          </button>
        </div>
      )}
      <form
        id="addWorkoutForm"
        onSubmit={handleSubmit}
        data-testid="addWorkoutForm"
      >
        <input
          type="text"
          className="search-input"
          name="workoutname"
          required
          placeholder="Workout Name"
          value={workoutName}
          onChange={handleWorkoutName}
        />
        <select
          className="search-input"
          name="muscletarget"
          required
          placeholder="Muscle Target"
          value={muscleTarget}
          onChange={handleMuscleTarget}
        >
          <option value="'" disabled>
            Muscle Target
          </option>
          <option value="Back">Back</option>
          <option value="Shoulders">Shoulders</option>
          <option value="Legs">Legs</option>
          <option value="Chest">Chest</option>
          <option value="Abs">Abs</option>
          <option value="Biceps">Biceps</option>
          <option value="Triceps">Triceps</option>
        </select>
        <input
          className="search-input"
          type="text"
          name="weight"
          required
          placeholder="Weight"
          value={weight}
          onChange={handleWeight}
        />
        <input
          className="search-input"
          type="text"
          name="reps"
          required
          placeholder="Reps"
          value={reps}
          onChange={handleReps}
        />
        {isPending ? (
          <div className="mini loading-pane">
            <h2 className="loader">ðŸ’ª</h2>
          </div>
        ) : (
          <div>
            <button className="button-theme" type="submit">
              Submit
            </button>
            <button
              className="button-theme"
              type="button"
              onClick={() => handleWorkoutModal()}
            >
              Cancel
            </button>
          </div>
        )}
      </form>
    </div>
  );
};
