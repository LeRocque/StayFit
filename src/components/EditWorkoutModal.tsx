import React, { ChangeEvent, FormEvent, useEffect, useState } from "react";
import { EditWorkoutModalProps, UserWorkoutsTypes } from "../frontendTypes";
import { useAppSelector } from "../hooks";
import { RootState } from "../store";

export const EditWorkoutModal = ({
  workoutId,
  handleEditModal,
}: EditWorkoutModalProps) => {
  const [workoutName, setWorkoutName] = useState("");
  const [muscleTarget, setMuscleTarget] = useState("");
  const [weight, setWeight] = useState("");
  const [reps, setReps] = useState("");
  const [errorMessage, setErrorMessage] = useState<string | null>(null);

  // access the store to get user workouts
  const workoutState = useAppSelector(
    (state: RootState) =>
      state.workouts.workouts as unknown as UserWorkoutsTypes[],
  );

  // if user has workouts, initialize workoutToEdit variable that matches workout id that was passed to our EditWorkoutModal props
  useEffect(() => {
    if (workoutState) {
      const workoutToEdit = workoutState.find(
        (el: UserWorkoutsTypes) => el.workout_id === workoutId,
      );
      // if there is a workout to edit, update name, muscleTarget, weight, and reps state to match workout to edit
      if (workoutToEdit) {
        setWorkoutName(workoutToEdit.workoutname);
        setMuscleTarget(workoutToEdit.muscletarget);
        setWeight(workoutToEdit.weight);
        setReps(workoutToEdit.reps);
      }
    }
    // update workout name, muscleTarget, weight, and reps any time workoutId prop changes or workoutState is updated
  }, [workoutId, workoutState]);

  // function that will make fetch request to our edit endpoint. We will send the passed down workoutId prop along with the current workoutName, muscleTarget, weight, and reps state
  const handleSubmit = async (e: FormEvent): Promise<void> => {
    e.preventDefault();
    try {
      const response = await fetch("/workout/edit", {
        method: "PUT",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          workout_id: workoutId,
          workoutName: workoutName,
          muscleTarget: muscleTarget,
          weight: weight,
          reps: reps,
        }),
      });
      // if workout was successfully editted we will update corresponding state values to empty strings and we will update handleEditModal to be null
      if (response.ok) {
        setWorkoutName("");
        setMuscleTarget("");
        setWeight("");
        setReps("");
        handleEditModal(null);
      } else {
        alert("invalid input");
      }
    } catch (err) {
      console.error(err);
      setErrorMessage("And error occurred while attempting to edit workout");
    }
  };

  // functions to update state with current values inputted by user in our form
  const handleWorkoutName = (e: ChangeEvent<HTMLSelectElement>): void =>
    setWorkoutName(e.target.value);
  const handleMuscleTarget = (e: ChangeEvent<HTMLSelectElement>): void =>
    setMuscleTarget(e.target.value);
  const handleWeight = (e: ChangeEvent<HTMLInputElement>): void =>
    setWeight(e.target.value);
  const handleReps = (e: ChangeEvent<HTMLInputElement>): void =>
    setReps(e.target.value);

  // if user clicks enter on modal div, we will close the modal
  const handleModalKeyPress = (
    e: React.KeyboardEvent<HTMLDivElement>,
  ): void => {
    if (e.key === "Enter") {
      handleEditModal(null);
    }
  };

  // we will create and render a div that will contain our form. If an error occurs we will render a div to describe the error to the user (this will also include a button to remove the message for the user). Our form will have input fields for workoutNames, muscleTargets, weight, and reps (workoutNames and muscleTargets will be selects and weight/reps will be text inputs)
  return (
    <div
      id="modal-container"
      onKeyDown={handleModalKeyPress}
      role="button"
      tabIndex={0}
    >
      {errorMessage && (
        <div className="error-message">
          {errorMessage}
          <button
            className="close-button"
            onClick={() => setErrorMessage(null)}
          >
            Dismiss Error
          </button>
        </div>
      )}
      <form
        className="mt-4 flex flex-col items-center justify-center"
        onSubmit={handleSubmit}
        data-testid="editModalForm"
      >
        <select
          className="search-input"
          name="workoutname"
          required
          placeholder="Workout Name"
          value={workoutName}
          onChange={handleWorkoutName}
        >
          {" "}
          <option value="'" disabled>
            Workout Name
          </option>
          <option value="Barbell Shrugs">Barbell Shrugs</option>
          <option value="Bench Dips">Bench Dips</option>
          <option value="Bench Press">Bench Press</option>
          <option value="Bicep Curl Barbell">Bicep Curl Barbell</option>
          <option value="Bicep Curl Dumbbells">Bicep Curl Dumbbells</option>
          <option value="Bicep Curl Pulley">Bicep Curl Pulley</option>
          <option value="Cable Flys">Cable Flys</option>
          <option value="Calf Raises">Calf Raises</option>
          <option value="Close Grip Bench">Close Grip Bench</option>
          <option value="Deadlifts">Deadlifts</option>
          <option value="Decline Bench Press">Decline Bench Press</option>
          <option value="Dips">Dips</option>
          <option value="Dumbbell Bench Alt">Dumbbell Bench Alt</option>
          <option value="Dumbbell Bench Press">Dumbbell Bench Press</option>
          <option value="Dumbbell Flys">Dumbbell Flys</option>
          <option value="Dumbbell Row">Dumbbell Row</option>
          <option value="Dumbbell Shrugs">Dumbbell Shrugs</option>
          <option value="Dumbbell Shoulder Press">
            Dumbbell Shoulder Press
          </option>
          <option value="Front Delt Raises">Front Delt Raises</option>
          <option value="Hammer Curls">Hammer Curls</option>
          <option value="Leg Lifts">Leg Lifts</option>
          <option value="Leg Press">Leg Press</option>
          <option value="Lunges">Lunges</option>
          <option value="Pull Ups">Pull Ups</option>
          <option value="Push Ups">Push Ups</option>
          <option value="Push Ups Alt">Push Ups Alt</option>
          <option value="Skull Crushers">Skull Crushers</option>
          <option value="Squat Alt">Squat Alt</option>
          <option value="Squats">Squats</option>
          <option value="Sit Ups">Sit Ups</option>
          <option value="Tricep Pulldowns">Tricep Pulldowns</option>
        </select>
        <select
          className="search-input"
          name="muscletarget"
          required
          placeholder="Muscle Target"
          value={muscleTarget}
          onChange={handleMuscleTarget}
        >
          <option value="'" disabled>
            Muscle Target
          </option>
          <option value="Abs">Abs</option>
          <option value="Back">Back</option>
          <option value="Biceps">Biceps</option>
          <option value="Chest">Chest</option>
          <option value="Legs">Legs</option>
          <option value="Shoulders">Shoulders</option>
          <option value="Triceps">Triceps</option>
        </select>
        <input
          className="search-input"
          type="text"
          name="weight"
          required
          placeholder="Weight"
          value={weight}
          onChange={handleWeight}
        />
        <input
          className="search-input"
          type="text"
          name="reps"
          required
          placeholder="Reps"
          value={reps}
          onChange={handleReps}
        />
        <button className="button-theme" type="submit">
          Submit
        </button>
        <button
          className="button-theme"
          type="button"
          onClick={() => handleEditModal(null)}
        >
          Cancel
        </button>
      </form>
    </div>
  );
};
