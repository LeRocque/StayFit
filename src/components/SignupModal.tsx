import { ChangeEvent, FormEvent, MouseEvent, useState } from "react";
import { SignupModalProps } from "../frontendTypes";

export const SignupModal = ({ handleModal }: SignupModalProps) => {
  const [email, setEmail] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [address, setAddress] = useState("");
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const handleSetEmail = (e: ChangeEvent<HTMLInputElement>) => {
    setEmail(e.target.value);
  };
  const handleSetFirstName = (e: ChangeEvent<HTMLInputElement>) => {
    setFirstName(e.target.value);
  };
  const handleSetLastName = (e: ChangeEvent<HTMLInputElement>) => {
    setLastName(e.target.value);
  };
  const handleSetAddress = (e: ChangeEvent<HTMLInputElement>) => {
    setAddress(e.target.value);
  };
  const handleSetUsername = (e: ChangeEvent<HTMLInputElement>) => {
    setUsername(e.target.value);
  };
  const handleSetPassword = (e: ChangeEvent<HTMLInputElement>) => {
    setPassword(e.target.value);
  };

  const handleModalClick = (e: MouseEvent<HTMLDivElement>) => {
    if (e.target === e.currentTarget) {
      handleModal();
    }
  };

  const handleSubmit = async (e: FormEvent) => {
    e.preventDefault();
    try {
      const response = await fetch("/user/signup", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        credentials: "include",
        body: JSON.stringify({
          email: email,
          firstName: firstName,
          lastName: lastName,
          address: address,
          username: username,
          password: password,
        }),
      });
      if (response.ok) {
        setEmail("");
        setFirstName("");
        setLastName("");
        setAddress("");
        setUsername("");
        setPassword("");
        handleModal();
      } else if (response.status === 401) {
        alert("username already exists, please select another");
        setUsername("");
      } else if (response.status === 400) {
        alert("all fields required");
      }
    } catch (err) {
      console.error(err);
    }
  };

  return (
    <div id="signupModalContainer" onClick={handleModalClick}>
      <form className="signup-form" onSubmit={handleSubmit}>
        <input
          type="text"
          required
          className="mb-5 block w-60"
          name="email"
          placeholder="Enter your email address"
          value={email}
          onChange={handleSetEmail}
        />
        <input
          type="text"
          required
          className="mb-5 block w-60"
          name="firstName"
          placeholder="Enter your first name"
          value={firstName}
          onChange={handleSetFirstName}
        />
        <input
          type="text"
          required
          className="mb-5 block w-60"
          name="lastName"
          placeholder="Enter your last name"
          value={lastName}
          onChange={handleSetLastName}
        />
        <input
          type="text"
          required
          className="mb-5 block w-60"
          name="address"
          placeholder="Enter your address"
          value={address}
          onChange={handleSetAddress}
        />
        <input
          type="text"
          required
          className="mb-5 block w-60"
          name="username"
          placeholder="Enter a username"
          value={username}
          onChange={handleSetUsername}
        />
        <input
          type="password"
          required
          className="mb-5 block w-60"
          name="password"
          placeholder="Enter a password"
          value={password}
          onChange={handleSetPassword}
        />
        <button
          className="m-2 rounded-full bg-blue-600 py-2 px-4 font-bold text-white hover:opacity-50"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  );
};
