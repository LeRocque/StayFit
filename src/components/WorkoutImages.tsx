import { WorkoutImages } from "../frontendTypes";

const GetWorkoutImages = (
  workoutName: string,
  workoutImages: WorkoutImages,
): string | undefined => {
  let imageResult;
  console.log("workoutImages are:", workoutImages);
  const cleanName = workoutName.replace(/[^a-zA-Z]/g, "").toLowerCase();
  if (cleanName === "situps") {
    imageResult = workoutImages.images.results[0].image;
  } else if (cleanName === "closegripbenchpress") {
    imageResult = workoutImages.images.results[3].image;
  } else if (cleanName === "bicepcurlpulley") {
    imageResult = workoutImages.images.results[4].image;
  } else if (cleanName === "bicepcurldumbbells") {
    imageResult = workoutImages.images.results[128].image;
  } else if (cleanName === "bicepcurlbarbell") {
    imageResult = workoutImages.images.results[6].image;
  } else if (cleanName === "benchdips") {
    imageResult = workoutImages.images.results[7].image;
  } else if (cleanName === "dumbbellshrugs") {
    imageResult = workoutImages.images.results[8].image;
  } else if (cleanName === "barbellshrugs") {
    imageResult = workoutImages.images.results[9].image;
  } else if (cleanName === "hammercurls") {
    imageResult = workoutImages.images.results[10].image;
  } else if (cleanName === "pushups") {
    imageResult = workoutImages.images.results[12].image;
  } else if (cleanName === "skullcrushers") {
    imageResult = workoutImages.images.results[13].image;
  } else if (cleanName === "benchpress") {
    imageResult = workoutImages.images.results[15].image;
  } else if (cleanName === "pullups") {
    imageResult = workoutImages.images.results[16].image;
  } else if (cleanName === "dumbbellbenchpress") {
    imageResult = workoutImages.images.results[25].image;
  } else if (cleanName === "declinebenchpress") {
    imageResult = workoutImages.images.results[26].image;
  } else if (cleanName === "cableflys") {
    imageResult = workoutImages.images.results[30].image;
  } else if (cleanName === "leglifts") {
    imageResult = workoutImages.images.results[33].image;
  } else if (cleanName === "squats") {
    imageResult = workoutImages.images.results[38].image;
  } else if (cleanName === "dumbbellshoulderpress") {
    imageResult = workoutImages.images.results[40].image;
  } else if (cleanName === "dips") {
    imageResult = workoutImages.images.results[46].image;
  } else if (cleanName === "dumbbellflys") {
    imageResult = workoutImages.images.results[47].image;
  } else if (cleanName === "squatstwo") {
    imageResult = workoutImages.images.results[52].image;
  } else if (cleanName === "frontdeltraises") {
    imageResult = workoutImages.images.results[74].image;
  } else if (cleanName === "pushupstwo") {
    imageResult = workoutImages.images.results[81].image;
  } else if (cleanName === "triceppulldown") {
    imageResult = workoutImages.images.results[86].image;
  } else if (cleanName === "lunges") {
    imageResult = workoutImages.images.results[106].image;
  } else if (cleanName === "legpress") {
    imageResult = workoutImages.images.results[107].image;
  } else if (cleanName === "dumbbellbenchpresstwo") {
    imageResult = workoutImages.images.results[92].image;
  } else if (cleanName === "calfraises") {
    imageResult = workoutImages.images.results[104].image;
  } else if (cleanName === "dumbbellrow") {
    imageResult = workoutImages.images.results[132].image;
  } else if (cleanName === "deadlifts") {
    imageResult = workoutImages.images.results[35].image;
  } else {
    imageResult = workoutImages.images.results[22].image;
  }
  return imageResult;
};

export default GetWorkoutImages;
