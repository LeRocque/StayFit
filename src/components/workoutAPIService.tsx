// import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";

// export const workoutApi = createApi({
//   reducerPath: "workoutApi",
//   baseQuery: fetchBaseQuery({
//     baseUrl: "https://wger.de/api/v2/",
//   }),
//   endpoints: (builder) => ({
//     getImages: builder.query({
//       query: (options) => `exerciseimage/?${options}`,
//       transformResponse: (response) => response.results,
//     }),
//   }),
// });

// export const { useGetImagesQuery } = workoutApi;
