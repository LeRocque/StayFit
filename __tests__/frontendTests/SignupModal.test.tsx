import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { SignupModal } from "../../src/components/SignupModal";

const modalProp = (): void => {
  throw new Error("Function not implemented.");
};

test("displays a signup form", async () => {
  const signupModal = render(<SignupModal handleModal={modalProp} />);
  const signupForm = await signupModal.findByTestId("signup-form");
  expect(signupForm).toBeDefined();
});

test("Should match its snapshot", () => {
  const signupModal = render(<SignupModal handleModal={modalProp} />);
  expect(signupModal).toMatchSnapshot();
});
