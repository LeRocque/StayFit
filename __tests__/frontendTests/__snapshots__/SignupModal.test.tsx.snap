// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="position-fixed fixed top-0 left-0 z-50 flex h-full w-full items-center justify-center bg-opacity-90 bg-gradient-to-br from-blue-200 via-blue-500 to-blue-700"
        role="button"
        tabindex="0"
      >
        <form
          class="signup-form"
          data-testid="signup-form"
        >
          <input
            class="search-input"
            name="email"
            placeholder="Enter your email"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="firstName"
            placeholder="Enter your first name"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="lastName"
            placeholder="Enter your last name"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="address"
            placeholder="Enter your address"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="username"
            placeholder="Enter a username"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="password"
            placeholder="Enter a password"
            required=""
            type="password"
            value=""
          />
          <button
            class="button-theme"
            type="submit"
          >
            Submit
          </button>
          <button
            class="button-theme"
            type="button"
          >
            Cancel
          </button>
        </form>
      </div>
    </div>
    <div>
      <div
        class="position-fixed fixed top-0 left-0 z-50 flex h-full w-full items-center justify-center bg-opacity-90 bg-gradient-to-br from-blue-200 via-blue-500 to-blue-700"
        role="button"
        tabindex="0"
      >
        <form
          class="signup-form"
          data-testid="signup-form"
        >
          <input
            class="search-input"
            name="email"
            placeholder="Enter your email"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="firstName"
            placeholder="Enter your first name"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="lastName"
            placeholder="Enter your last name"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="address"
            placeholder="Enter your address"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="username"
            placeholder="Enter a username"
            required=""
            type="text"
            value=""
          />
          <input
            class="search-input"
            name="password"
            placeholder="Enter a password"
            required=""
            type="password"
            value=""
          />
          <button
            class="button-theme"
            type="submit"
          >
            Submit
          </button>
          <button
            class="button-theme"
            type="button"
          >
            Cancel
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="position-fixed fixed top-0 left-0 z-50 flex h-full w-full items-center justify-center bg-opacity-90 bg-gradient-to-br from-blue-200 via-blue-500 to-blue-700"
      role="button"
      tabindex="0"
    >
      <form
        class="signup-form"
        data-testid="signup-form"
      >
        <input
          class="search-input"
          name="email"
          placeholder="Enter your email"
          required=""
          type="text"
          value=""
        />
        <input
          class="search-input"
          name="firstName"
          placeholder="Enter your first name"
          required=""
          type="text"
          value=""
        />
        <input
          class="search-input"
          name="lastName"
          placeholder="Enter your last name"
          required=""
          type="text"
          value=""
        />
        <input
          class="search-input"
          name="address"
          placeholder="Enter your address"
          required=""
          type="text"
          value=""
        />
        <input
          class="search-input"
          name="username"
          placeholder="Enter a username"
          required=""
          type="text"
          value=""
        />
        <input
          class="search-input"
          name="password"
          placeholder="Enter a password"
          required=""
          type="password"
          value=""
        />
        <button
          class="button-theme"
          type="submit"
        >
          Submit
        </button>
        <button
          class="button-theme"
          type="button"
        >
          Cancel
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
